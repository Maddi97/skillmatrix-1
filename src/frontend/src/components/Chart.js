import React, { Component } from "react";
import { Line } from "react-chartjs-2";
import * as zoom from "chartjs-plugin-zoom";

// example for "Skill" in json format

class Chart extends Component {
  render() {
    const data = {
      datasets: [
        {
          label: this.props.skill,
          data: this.props.data,
          pointBorderColor: "rgba(75,192,192,1)",
          borderColor: "rgba(75,192,192,1)",
          pointHoverBackgroundColor: "rgba(75,192,192,1)",
          pointHoverBorderColor: "rgba(75,192,192,1)",
          backgroundColor: "rgba(75,192,192,1)",
          fill: false,
          pointBorderWidth: "rgba(75,192,192,1)"
        }
      ]
    };
    return (
      <div className="Chart">
        <Line
          data={data}
          width={this.props.width}
          height={this.props.height}
          options={{
            scales: {
              xAxes: [
                {
                  type: "time",
                  position: "bottom",
                  ticks: { display: this.props.display },
                  time: {
                    displayFormats: { month: "MM/YY" },
                    tooltipFormat: "ll",
                    unit: "month"
                  }
                }
              ],
              yAxes: [
                {
                  ticks: {
                    display: this.props.display,
                    beginAtZero: true,
                    steps: 0.5,
                    stepValue: 1,
                    max: 7
                  }
                }
              ]
            },
            pan: {
              enabled: this.props.enabledZoom,
              mode: "x"
            },
            zoom: {
              enabled: this.props.enabledZoom,
              mode: "xy"
            },
            startAtZero: true,
            maintainAspectRatio: false
          }}
        />
      </div>
    );
  }
}

export default Chart;
