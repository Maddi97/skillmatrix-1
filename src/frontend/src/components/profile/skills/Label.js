import React from 'react';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';

const styles = theme => ({
  root: {
    padding: theme.spacing.unit * 2,
    marginTop: theme.spacing.unit,
    textAlign: 'left',
  },
});

function Label(props) {
  const { classes, value } = props;

  return (
    <div>
      <Paper className={classes.root} elevation={1}>
        <Typography variant="h5" component="h3">
          {value}
        </Typography>
        <Typography component="p">{value}</Typography>
      </Paper>
    </div>
  );
}

Label.propTypes = {
  classes: PropTypes.instanceOf(styles).isRequired,
};

export default withStyles(styles)(Label);
