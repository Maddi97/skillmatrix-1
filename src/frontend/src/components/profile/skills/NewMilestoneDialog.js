import React from 'react';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import { DateInput, TextArea, LevelPicker } from 'components/common/InputFields';
import SingleSelect from 'components/common/SingleSelect';

// redux
import store from 'Store';
import { closeProfileDialog, updateInput, resetForm } from 'actions';

export default class FormDialog extends React.Component {
  handleClickOpen = () => {
    this.setState({ open: true });
  };

  handleClose = () => {
    store.dispatch(resetForm);
    store.dispatch(closeProfileDialog);
  };

  handleSubmit = () => {
    console.log('submit');
    this.handleClose();
  };

  handleChange(id, value) {
    store.dispatch(updateInput(id, value));
  }

  render() {
    const state = store.getState();
    const { showDialog } = state.profile;
    const { datefield, textarea, levelfield } = state.formState;
    return (
      <div>
        <Dialog
          open={showDialog === 'milestone'}
          onClose={this.handleClose}
          aria-labelledby="form-dialog-title"
        >
          <DialogTitle id="form-dialog-title">New milestone</DialogTitle>
          <DialogContent>
            <DialogContentText>
              To add a new milestone please fill in all inputfields.
            </DialogContentText>
            <SingleSelect />
            <LevelPicker data={levelfield} onChange={(id, value) => this.handleChange(id, value)} />
            <DateInput data={datefield} onChange={(id, value) => this.handleChange(id, value)} />
            <TextArea data={textarea} onChange={(id, value) => this.handleChange(id, value)} />
          </DialogContent>
          <DialogActions>
            <Button onClick={this.handleClose} color="primary">
              Cancel
            </Button>
            <Button onClick={this.handleSubmit} color="primary">
              Submit
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    );
  }
}
