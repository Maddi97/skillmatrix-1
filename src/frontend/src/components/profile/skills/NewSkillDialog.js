import React from 'react';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import { DateInput, TextArea, LevelPicker } from 'components/common/InputFields';

import SingleSelect from 'components/common/SingleSelect';

// redux
import store from 'Store';
import { changeView, openProfileDialog, closeProfileDialog, updateInput, resetForm } from 'actions';

export default class FormDialog extends React.Component {
  handleClickOpen = () => {
    this.setState({ open: true });
  };

  handleClose = () => {
    store.dispatch(resetForm);
    store.dispatch(closeProfileDialog);
  };

  handleSubmit = skill => {
    console.log(skill);
    //skill objekt an api Ã¼bergenen

    this.handleClose();
  };

  handleChange(id, value) {
    store.dispatch(updateInput(id, value));
  }

  render() {
    const state = store.getState();

    const { showDialog } = state.profile;
    var { datefield, textarea, levelfield, singleselect } = state.formState;

    var profile = state.profile.profiles[state.profile.person];
    var newSkill = '';
    const aktMilestone = [
      {
        datum: datefield.value,
        level: levelfield.value,
        comment: textarea.value,
      },
    ];

    const aktSkill = {
      username: profile.username,
      skills: { skillname: singleselect.value, level: levelfield.value, milestones: aktMilestone },
    };
    const allSkillsOfUser = [];
    Object.keys(profile.skills).map(element => {
      allSkillsOfUser[element] = profile.skills[element].skillname;
    });

    const availableNewSkills = [];

    Object.keys(state.allSkills).map(index => {
      if (!allSkillsOfUser.includes(state.allSkills[index])) {
        availableNewSkills.push(state.allSkills[index]);
      }
    });

    return (
      <div>
        <Dialog
          open={showDialog === 'skill'}
          onClose={this.handleClose}
          aria-labelledby="form-dialog-title"
        >
          <DialogTitle id="form-dialog-title">New skill</DialogTitle>
          <DialogContent>
            <DialogContentText>
              To add a new Skill please fill in all inputfields.
            </DialogContentText>
            <SingleSelect allSkills={availableNewSkills} />
            <TextField
              id="standard-with-placeholder"
              label="With placeholder"
              placeholder="Placeholder"
              margin="normal"
              onChange={event => (newSkill = event.target.value)}
            />

            <Button onClick={() => (singleselect.value = newSkill)}>+</Button>
            <LevelPicker data={levelfield} onChange={(id, value) => this.handleChange(id, value)} />
            <DateInput data={datefield} onChange={(id, value) => this.handleChange(id, value)} />
            <TextArea data={textarea} onChange={(id, value) => this.handleChange(id, value)} />
          </DialogContent>
          <DialogActions>
            <Button onClick={this.handleClose} color="primary">
              Cancel
            </Button>
            <Button onClick={() => this.handleSubmit(aktSkill)} color="primary">
              Submit
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    );
  }
}
