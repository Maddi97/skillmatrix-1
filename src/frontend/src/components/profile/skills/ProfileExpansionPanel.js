//react
import React from 'react';

//redux
import store from 'Store';

// material-ui
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import Typography from '@material-ui/core/Typography';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import RadioGroup from '../../common/RadioGroup';
import Button from '@material-ui/core/Button';

const styles = theme => ({
  heading: {
    display: 'flex',
    flexDirection: 'column',
  },

  secondaryHeading: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'left',
    alignItems: 'stretch',
    flexBasis: '99%',
    color: theme.palette.text.secondary,
  },
  row: { width: '320%' },
});

class ControlledExpansionPanels extends React.Component {
  state = {
    expanded: null,
  };
  handleChange = panel => (event, expanded) => {
    this.setState({
      expanded: expanded ? panel : false,
    });
  };

  getGuidelines(skillname, allSkills) {
      if (skillname in allSkills) return allSkills[skillname];
    
  }

  render() {
    const { classes, skill, summary } = this.props;
    const { skillname, level, skillpath } = skill;
    const { expanded } = this.state;
    const { allSkills } = store.getState();
    const guidelines = this.getGuidelines(skillpath, allSkills);
    return (
      <div className={classes.root}>
        {summary.length !== 0 ? (
          <ExpansionPanel
            expanded={expanded === skill}
            onChange={this.handleChange(skill)}
          >
            <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
              <Typography component={'span'} className={classes.Heading}>
                <div>
                  <div>
                    <Button
                      variant="outlined"
                      color="primary"
                      style={{ textTransform: 'none' }}
                      fullWidth
                    >
                      {skillname}
                    </Button>
                  </div>
                  <div className={classes.row} onClick={event => event.stopPropagation()}>
                    <RadioGroup
                      level={level}
                      skill={skillpath}
                      levelChange={this.props.levelChange}
                      disabled={this.props.isEditable}
                      guidelines={guidelines}
                    />
                  </div>
                </div>
              </Typography>
            </ExpansionPanelSummary>
            <ExpansionPanelDetails className={classes.secondaryHeading}>
              {' '}
              {this.props.summary}
            </ExpansionPanelDetails>
          </ExpansionPanel>
        ) : (
          <ExpansionPanel>
            <ExpansionPanelSummary>
              <Typography component={'span'} className={classes.Heading}>
                <div>
                  <div>
                    <Button
                      variant="outlined"
                      color="primary"
                      style={{ textTransform: 'none' }}
                      fullWidth
                    >
                      {skillname}
                    </Button>
                  </div>
                  <div className={classes.row} onClick={event => event.stopPropagation()}>
                    <RadioGroup
                      level={level}
                      skill={skillpath}
                      levelChange={this.props.levelChange}
                      disabled={this.props.isEditable}
                      guidelines={guidelines}
                    />
                  </div>
                </div>
              </Typography>
            </ExpansionPanelSummary>
          </ExpansionPanel>
        )}
      </div>
    );
  }
}

ControlledExpansionPanels.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(ControlledExpansionPanels);
