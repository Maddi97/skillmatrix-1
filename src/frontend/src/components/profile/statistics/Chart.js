import React, { Component } from 'react';
import { Line } from 'react-chartjs-2';

// example for "Skill" in json format

class Chart extends Component {
  render() {
    const { skill, data, witdh, height, display, enabledZoom } = this.props;
    const dataSet = {
      datasets: [
        {
          label: skill,
          data: data,
          pointBorderColor: 'rgba(75,192,192,1)',
          borderColor: 'rgba(75,192,192,1)',
          pointHoverBackgroundColor: 'rgba(75,192,192,1)',
          pointHoverBorderColor: 'rgba(75,192,192,1)',
          backgroundColor: 'rgba(75,192,192,1)',
          fill: false,
          pointBorderWidth: 'rgba(75,192,192,1)',
        },
      ],
    };

    console.log(dataSet);
    return (
      <div className="Chart">
        <Line
          data={dataSet}
          width={witdh}
          height={height}
          options={{
            scales: {
              xAxes: [
                {
                  type: 'time',
                  position: 'bottom',
                  ticks: { display },
                  time: {
                    displayFormats: { month: 'MM/YY' },
                    tooltipFormat: 'll',
                    unit: 'month',
                  },
                },
              ],
              yAxes: [
                {
                  ticks: {
                    display,
                    beginAtZero: true,
                    steps: 0.5,
                    stepValue: 1,
                    max: 7,
                  },
                },
              ],
            },
            pan: {
              enabled: enabledZoom,
              mode: 'x',
            },
            zoom: {
              enabled: enabledZoom,
              mode: 'xy',
            },
            startAtZero: true,
            maintainAspectRatio: false,
          }}
        />
      </div>
    );
  }
}

export default Chart;
