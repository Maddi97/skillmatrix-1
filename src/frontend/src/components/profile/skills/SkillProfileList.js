import React from 'react';
import ProfileExpansionPanel from 'components/profile/skills/ProfileExpansionPanel';

export default (props) => {
  const { categories } = props;
  // console.log(categories);
  const panels = categories.map(skill => (
    <div>
      <ProfileExpansionPanel skill={skill} levelChange={props.levelChange} />
    </div>
  ));
  return panels;
};
