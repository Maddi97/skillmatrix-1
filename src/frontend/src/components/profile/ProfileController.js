// react
import React, { Component } from 'react';

// redux
import store from 'Store';
import { changeView, openProfileDialog, switchPage, setOwnProfile, setError } from 'actions';

// react components
import TabContainer from 'components/profile/TabContainer';
import SkillProfileList from 'components/profile/skills/SkillProfileList';
import SkillStatisticsGrid from 'components/profile/statistics/SkillStatisticsGrid';
import NewMilestoneDialog from 'components/profile/skills/NewMilestoneDialog';
import NewSkillDialog from 'components/profile/skills/NewSkillDialog';

// material-ui
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import Button from '@material-ui/core/Button';
import ButtonBase from '@material-ui/core/ButtonBase';
import IconButton from '@material-ui/core/IconButton';
import { ArrowLeft, Search } from '@material-ui/icons';

// Rest
import RestPoints from 'rest/Init';
import RestCom from 'rest/Rest';
import { updateAllSkills } from 'rest/handleCommonRequests';

const styles = theme => ({
  root: {
    flexGrow: 1,
    backgroundColor: theme.palette.background.paper,
  },

  buttons: {
    margin: 0,
    top: 'auto',
    right: 75,
    bottom: 50,
    left: 'auto',
    position: 'fixed',
  },
});

class ProfileController extends Component {
  state = {
    changes: false,
  };
  localUpdate = [];

  async applyLevelUpdates(skills) {
    const { state } = this.props;
    const { person } = state.profile;
    const { username } = state.profile.profiles[person];
    var latestChanges = { username, skills: {} };
    var alreadyUpdated = [];
    Object.keys(skills).map(index => {
      Object.keys(this.localUpdate).map(idx => {
        if (skills[index].skillpath === this.localUpdate[idx][0].skill) {
          alreadyUpdated.push(this.localUpdate[idx][0].skill);
          latestChanges.skills[this.localUpdate[idx][0].skill] = this.localUpdate[idx][0].level;
        }
      });
    });
    console.log(latestChanges);
    // send skill
    let Rest = new RestCom(RestPoints.setSkills, latestChanges);

    try {
      const newProfile = await Rest.post();
      store.dispatch(setOwnProfile(newProfile));
    } catch (e) {
      store.dispatch(setError(e.message));
    }

    this.localUpdate = [];
    this.setState({ changes: false });
    delete latestChanges[skills];
  }

  handleLevelChange = (skill, level) => {
    this.localUpdate.push([{ skill, level }]);
    this.setState({ changes: true });
  };

  getOwnerArticle() {
    const { state } = this.props;
    const { person, isEditable } = state.profile;
    const { username } = state.profile.profiles[person];

    if (isEditable) {
      return 'My ';
    }

    return username + "'s ";
  }

  handleChange = (evt, value) => {
    store.dispatch(changeView(value));
  };

  getAllSkillsRecursive(skill, allSkills) {
    allSkills.push(skill);
    skill.subcategories.map(subskill => {
      this.getAllSkillsRecursive(subskill, allSkills);
    });
    return allSkills;
  }

  async openNewSkillDialog() {
    await updateAllSkills();
    store.dispatch(openProfileDialog('skill'));
  }

  async openNewMilestoneDialog() {
    await updateAllSkills();
    store.dispatch(openProfileDialog('milestone'));
  }

  switchToSearchPage = () => {
    store.dispatch(switchPage('search'));
  };

  render() {
    const { state, classes } = this.props;
    // person: array index in profiles
    const { person, profiles, view, showDialog, isEditable } = state.profile;
    const skills = profiles[person].skills;

    var allSkillsOfUser = [];
    Object.keys(profiles[person].skills).map(index => {
      Object.keys(profiles[person].skills[index].subcategories).map(subskill => {
        if (allSkillsOfUser.length === 0) {
          allSkillsOfUser = this.getAllSkillsRecursive(
            profiles[person].skills[index].subcategories[subskill],
            [],
          );
        } else
          allSkillsOfUser.push(
            ...this.getAllSkillsRecursive(
              profiles[person].skills[index].subcategories[subskill],
              [],
            ),
          );
      });
    });

    //console.log(profiles[person]);
    var skillsTmp = [];
    try {
      skillsTmp = profiles[person].skills;
    } catch {
      return <h2>Could not load profile.</h2>;
    }
    const ownerArticle = this.getOwnerArticle();
    return (
      <div className={classes.root}>
        <AppBar position="static">
          <Tabs value={view} onChange={this.handleChange} centered fullWidth>
            <Tab label={ownerArticle + 'Profile'} />
            <Tab label={ownerArticle + 'Statistics'} />
          </Tabs>
        </AppBar>
        {!isEditable && (
          <IconButton
            className={classes.goBackButton}
            onClick={this.switchToSearchPage}
            color="inherit"
            aria-label="Go Back"
          >
            <ArrowLeft />
            <Search />
          </IconButton>
        )}
        <div>
          {view === 0 && (
            <TabContainer>
              <SkillProfileList
                categories={skills}
                levelChange={this.handleLevelChange}
                isEditable={isEditable}
              />

              {isEditable && (
                <div>
                  <NewMilestoneDialog open={showDialog === 'milestone'} />
                  <NewSkillDialog />
                  <div className={classes.buttons}>
                    <Button
                      variant="contained"
                      color="primary"
                      onClick={this.openNewMilestoneDialog}
                    >
                      New Milestone
                    </Button>
                    <Button
                      variant="contained"
                      color="primary"
                      name="submit"
                      onClick={this.openNewSkillDialog}
                    >
                      New Skill
                    </Button>
                    {this.state.changes && (
                      <Button
                        className="applyButton"
                        variant="contained"
                        color="secondary"
                        onClick={this.applyLevelUpdates.bind(this, allSkillsOfUser)}
                      >
                        Apply Changes
                      </Button>
                    )}
                  </div>
                </div>
              )}
            </TabContainer>
          )}
          {view === 1 && (
            <TabContainer>
              <SkillStatisticsGrid categories={skills} />
            </TabContainer>
          )}
        </div>
      </div>
    );
  }
}

ProfileController.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(ProfileController);
