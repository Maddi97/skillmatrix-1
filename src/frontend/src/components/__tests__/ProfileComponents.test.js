import React from 'react';
import { shallow } from 'enzyme';

// import profile/skills components

import Label from '../profile/skills/Label';
import NewMilestoneDialog from '../profile/skills/NewMilestoneDialog';
import NewSkillDialog from '../profile/skills/NewSkillDialog';
import ProfileExpansionPanel from '../profile/skills/ProfileExpansionPanel';
import SkillProfileList from '../profile/skills/SkillProfileList';

// import profile/statistics components

import Chart from '../profile/statistics/Chart';
import SimpleCard from '../profile/statistics/SimpleCard';
import SkillStatisticsGrid from '../profile/statistics/Chart';

// import profile main components

import ProfileController from '../profile/ProfileController';
import TabContainer from '../profile/TabContainer';

describe('test Profile/skills Components', () => {
  it('render Label component', () => {
    const wrapper = shallow(<Label value="Javascript" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('render NewMilestoneDialog component', () => {
    const wrapper = shallow(<NewMilestoneDialog open />);
    expect(wrapper).toMatchSnapshot();
  });
  it('render NewSkillDialog component', () => {
    const wrapper = shallow(<NewSkillDialog />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render SkillProfileList component', () => {
    let wrapper = shallow(
      <SkillProfileList
        categories={[{ skillname: 'Java' }, { skillname: 'Python' }, { skillname: 'Haskell' }]}
        isEditable
      />,
    );
    expect(wrapper).toMatchSnapshot();

    wrapper = shallow(
      <SkillProfileList
        categories={[{ skillname: 'Java' }, { skillname: 'Python' }, { skillname: 'Haskell' }]}
        isEditable={false}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('render ProfileExpensionPanel component', () => {
    let wrapper = shallow(<ProfileExpansionPanel skill="Javascript" key={1} isEditable />);
    expect(wrapper).toMatchSnapshot();

    wrapper = shallow(<ProfileExpansionPanel skill="Javascript" key={1} isEditable={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('test Profile/stats Components', () => {
  it('render Chart component', () => {
    const wrapper = shallow(<Chart skill="Javascript" data={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('render SimpleCard component', () => {
    const wrapper = shallow(<SimpleCard skill="Javascript" data={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('render SkillStatisticsGrid component', () => {
    const wrapper = shallow(<SkillStatisticsGrid skill="Javascript" data={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('test Profile Components', () => {
  it('render ProfileController component', () => {
    const wrapper = shallow(<ProfileController skill="Javascript" data={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('render TabContainer component', () => {
    const wrapper = shallow(<TabContainer skill="Javascript" children={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
