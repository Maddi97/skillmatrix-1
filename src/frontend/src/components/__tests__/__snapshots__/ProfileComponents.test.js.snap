// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Profile Components render ProfileController component 1`] = `
<ProfileController
  classes={
    Object {
      "buttonMargin": "ProfileController-buttonMargin-14",
      "buttons": "ProfileController-buttons-13",
      "root": "ProfileController-root-12",
    }
  }
  data={Array []}
  skill="Javascript"
/>
`;

exports[`test Profile Components render TabContainer component 1`] = `
<TabContainer
  classes={
    Object {
      "root": "TabContainer-root-15",
    }
  }
  skill="Javascript"
/>
`;

exports[`test Profile/skills Components render Label component 1`] = `
<Label
  classes={
    Object {
      "root": "Label-root-1",
    }
  }
  value="Javascript"
/>
`;

exports[`test Profile/skills Components render NewMilestoneDialog component 1`] = `
<div>
  <WithStyles(Dialog)
    aria-labelledby="form-dialog-title"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(DialogTitle)
      id="form-dialog-title"
    >
      New milestone
    </WithStyles(DialogTitle)>
    <WithStyles(DialogContent)>
      <WithStyles(DialogContentText)>
        To add a new milestone please fill in all inputfields.
      </WithStyles(DialogContentText)>
      <WithStyles(IntegrationReactSelect)
        allSkills={
          Array [
            "Programming/Python",
            "Programming/Python/NumPy",
            "Design/Photoshop",
            "Sport/Aikido",
            "Sport/Socker",
          ]
        }
      />
      <WithStyles(RadioButtonsGroup)
        level={0}
      />
      <WithStyles(Component)
        data={
          Object {
            "error": false,
            "name": "Date",
            "value": "2019-01-11",
          }
        }
        onChange={[Function]}
      />
      <WithStyles(Component)
        data={
          Object {
            "error": false,
            "name": "Note",
            "value": "",
          }
        }
        onChange={[Function]}
      />
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        color="primary"
        onClick={[Function]}
      >
        Cancel
      </WithStyles(Button)>
      <WithStyles(Button)
        color="primary"
        onClick={[Function]}
      >
        Submit
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</div>
`;

exports[`test Profile/skills Components render NewSkillDialog component 1`] = `
<div>
  <WithStyles(Dialog)
    aria-labelledby="form-dialog-title"
    onClose={[Function]}
  >
    <WithStyles(DialogTitle)
      id="form-dialog-title"
    >
      New skill
    </WithStyles(DialogTitle)>
    <WithStyles(DialogContent)>
      <WithStyles(DialogContentText)>
        To add a new Skill please fill in all inputfields.
      </WithStyles(DialogContentText)>
      <WithStyles(IntegrationReactSelect)
        allSkills={
          Array [
            "Programming/Python/SQLAlchemy",
            "Programming/Java",
            "Design/Skillname",
            "Sport/Kung Fu",
          ]
        }
        placeholder="Select a skill to add"
      />
      <WithStyles(Component)
        data={
          Object {
            "error": false,
            "name": "Level",
            "value": "",
          }
        }
        onChange={[Function]}
        required={true}
      />
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        color="primary"
        onClick={[Function]}
      >
        Cancel
      </WithStyles(Button)>
      <WithStyles(Button)
        color="primary"
        onClick={[Function]}
      >
        Submit
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</div>
`;

exports[`test Profile/skills Components render ProfileExpensionPanel component 1`] = `
<ControlledExpansionPanels
  classes={
    Object {
      "headingButton": "ControlledExpansionPanels-headingButton-5",
      "left": "ControlledExpansionPanels-left-3",
      "noFlex": "ControlledExpansionPanels-noFlex-7",
      "removeButton": "ControlledExpansionPanels-removeButton-6",
      "right": "ControlledExpansionPanels-right-4",
    }
  }
  isEditable={true}
  skill="Javascript"
/>
`;

exports[`test Profile/skills Components render ProfileExpensionPanel component 2`] = `
<ControlledExpansionPanels
  classes={
    Object {
      "headingButton": "ControlledExpansionPanels-headingButton-5",
      "left": "ControlledExpansionPanels-left-3",
      "noFlex": "ControlledExpansionPanels-noFlex-7",
      "removeButton": "ControlledExpansionPanels-removeButton-6",
      "right": "ControlledExpansionPanels-right-4",
    }
  }
  isEditable={false}
  skill="Javascript"
/>
`;

exports[`test Profile/skills Components render SkillProfileList component 1`] = `
<SkillProfileList
  categories={
    Array [
      Object {
        "skillname": "Java",
      },
      Object {
        "skillname": "Python",
      },
      Object {
        "skillname": "Haskell",
      },
    ]
  }
  classes={
    Object {
      "panels": "SkillProfileList-panels-2",
    }
  }
  isEditable={true}
/>
`;

exports[`test Profile/skills Components render SkillProfileList component 2`] = `
<SkillProfileList
  categories={
    Array [
      Object {
        "skillname": "Java",
      },
      Object {
        "skillname": "Python",
      },
      Object {
        "skillname": "Haskell",
      },
    ]
  }
  classes={
    Object {
      "panels": "SkillProfileList-panels-2",
    }
  }
  isEditable={false}
/>
`;

exports[`test Profile/stats Components render Chart component 1`] = `
<div
  className="Chart"
>
  <Line
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": "rgba(75,192,192,1)",
            "borderColor": "rgba(75,192,192,1)",
            "data": Array [
              Object {
                "x": "2011-09-11",
                "y": 0,
              },
              Object {
                "x": "2017-01-20",
                "y": 1,
              },
              Object {
                "x": "2018-02-06",
                "y": 2,
              },
              Object {
                "x": "2021-11-23",
                "y": 3,
              },
            ],
            "fill": false,
            "label": "C++",
            "pointBorderColor": "rgba(75,192,192,1)",
            "pointBorderWidth": "rgba(75,192,192,1)",
            "pointHoverBackgroundColor": "rgba(75,192,192,1)",
            "pointHoverBorderColor": "rgba(75,192,192,1)",
          },
        ],
      }
    }
    options={
      Object {
        "maintainAspectRatio": false,
        "pan": Object {
          "enabled": undefined,
          "mode": "x",
        },
        "scales": Object {
          "xAxes": Array [
            Object {
              "position": "bottom",
              "ticks": Object {
                "display": undefined,
              },
              "time": Object {
                "displayFormats": Object {
                  "month": "MM/YY",
                },
                "tooltipFormat": "ll",
                "unit": "month",
              },
              "type": "time",
            },
          ],
          "yAxes": Array [
            Object {
              "ticks": Object {
                "beginAtZero": true,
                "display": undefined,
                "max": 7,
                "stepValue": 1,
                "steps": 0.5,
              },
            },
          ],
        },
        "startAtZero": true,
        "zoom": Object {
          "enabled": undefined,
          "mode": "xy",
          "rangeMax": Object {
            "y": 30,
          },
          "rangeMin": Object {
            "y": -10,
          },
        },
      }
    }
  />
</div>
`;

exports[`test Profile/stats Components render SimpleCard component 1`] = `
<ClickableChart
  classes={
    Object {
      "appBar": "ClickableChart-appBar-9",
      "container": "ClickableChart-container-11",
      "flex": "ClickableChart-flex-10",
      "root": "ClickableChart-root-8",
    }
  }
  milestones={
    Array [
      Object {
        "comment": "init",
        "date": "2011-09-11",
        "level": 0,
      },
      Object {
        "comment": "C++ Workshop",
        "date": "2017-01-20",
        "level": 1,
      },
      Object {
        "comment": "C++ Lehrgang",
        "date": "2018-02-06",
        "level": 2,
      },
      Object {
        "comment": "C++ 3 jÃ¤hriges Projekt fertig gestellt, mit 100000 Zeilen c++ Code",
        "date": "2021-11-23",
        "level": 3,
      },
    ]
  }
  skillname="C++"
  skillpath="C++"
/>
`;
